src/amo/components/SponsoredAddonsShelf/index.js

```diff
@@ -68,8 +83,37 @@ export class SponsoredAddonsShelfBase extends React.Component<InternalProps> {
         }),
       );
     }
+
+    this.sendImpressionPing();
+  }
+
+  componentDidUpdate(prevProps: InternalProps) {
+    const { shelfData } = this.props;
+
+    if (prevProps.shelfData !== shelfData) {
+      this.sendImpressionPing();
+    }
   }
 
+  sendImpressionPing = () => {
+    const { _config, _navigator, shelfData } = this.props;
+
+    if (_config.get('enableFeatureUseAdzerkForSponsoredShelf') && shelfData) {
+      const { impressionData, impressionURL } = shelfData;
+
+      if (impressionData && impressionURL) {
```

Comment:
Should we log a debug message (at least) when we don't have data/url?

Reply:
Sure, good idea.